### PR TITLE
Add msi build scripts

### PR DESCRIPTION
**Pull request by twangboy**
### What does this PR do?
Brings the MSI build scripts from https://github.com/saltstack/salt-windows-msi to Salt

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [ ] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [ ] Tests written/updated

### Commits signed with GPG?
Yes

### CODE REVIEW

**Comment by Ch3LL** [comment]
pre-commit is failing and there is a merge conflict.

File: pkg/windows/msi/tests/config_tests/.gitignore [hunk 1]
```diff
@@ -0,0 +1,4 @@
+*.bat
+*.log
+*.output
+
```
Comment (Ch3LL) [inline pkg/windows/msi/tests/config_tests/.gitignore:4]: remove blank line

File: requirements/windows.txt [hunk 2]
```diff
@@ -4,7 +4,7 @@
 
 pywin32>=305
 wmi>=1.5.1
-pythonnet>=2.5.2
+pythonnet>=3.0.1
```
Comment (Ch3LL) [inline requirements/windows.txt:7]: It seems like there are multiple things occuring in this PR.
- Adding MSI build scripts
- Updating pythonnet version. 
Can we break the PRs out so we don't have one big PR?

**Review by Ch3LL — CHANGES_REQUESTED** [review]
Why are we adding binaries to a repo? such as python.exe, binlevel.exe, etc?

**Comment by Ch3LL** [comment]
Looks like you did make a separate PR for pythonnet upgrade: https://github.com/saltstack/salt/pull/63355

Can you remove the change from here?

File: pkg/windows/msi/tests/config_tests/.gitignore (hunk 1 above)
Comment (twangboy) [reply pkg/windows/msi/tests/config_tests/.gitignore]: Done

File: requirements/windows.txt (hunk 2 above)
Comment (twangboy) [reply requirements/windows.txt]: The pythonnet PR has been merged and this rebased.

**Comment by Ch3LL** [comment]
Why are we adding binaries to a repo? such as python.exe, binlevel.exe, etc?

**Comment by twangboy** [comment]
Most of the binaries are just junk... required by the installer code because it's expecting a file with that name. The Python binaries are required because the build script runs the python to get the arch it was built for. In a normal environment this would be the Python.exe/.dll created by relenve. For the test environment, we don't really care. It just needs to return the arch of the system.

**Review by garethgreenaway — APPROVED** [review]

**Pull request merged**